### ISSUE TITLE
feature: Use class name by default for `@DgsDataLoader#name`, and add some inline methods to `DgsDataFetchingEnvironment`

### ISSUE THREAD

**Issue by jord1e** [Netflix/dgs-framework#861]
~~Please read our [contributor guide](https://github.com/Netflix/dgs-framework/blob/master/CONTRIBUTING.md) before
creating an issue. Also consider discussing your idea on
the [discussion forum](https://github.com/Netflix/dgs-framework/discussions) first.~~

## Describe the Feature Request
The the annotated class's name as the `name` field of the `@DgsDataLoader` annotation __by default__
When annotating methods use something like `SimpleClassName.methodName`

## Describe Preferred Solution
Using
```kotlin
@DgsDataLoader
class PersonByIdDataLoader : MappedBatchLoader<String, Try<Person>>{
    override fun load(keys: Set<String>): CompletionStage<Map<String, Try<Person>>> {
        TODO("Not yet implemented")
    }
}
```

We can then do:
```kotlin
val dl = dfe.getDataLoader<String, Try<Person>>(PersonByIdDataLoader::class.java)
// or
val dl = dfe.getDataLoader<String, Try<Person>>("PersonByIdDataLoader")
```

We may also wish to add some extra (inline-reified) methods to DgsDataFetchingEnvironment, for example:
```kotlin
inline fun <K, V, reified T : MappedBatchLoader<K, V>> DgsDataFetchingEnvironment.getMappedBatchLoader(): DataLoader<K, V> {
    return this.getDataLoader(T::class.java)
}
inline fun <K, V, reified T : BatchLoader<K, V>> DgsDataFetchingEnvironment.getBatchLoader(): DataLoader<K, V> {
    return this.getDataLoader(T::class.java)
}
```

Which gives us compile-time type checking:
incorrect:
![afbeelding](https://user-images.githubusercontent.com/30464310/152649360-30da0d63-2165-4cfa-adae-e68c3791c3e4.png)

correct:
![afbeelding](https://user-images.githubusercontent.com/30464310/152649642-98ceedb6-4843-4c9a-aba9-94a2d8829a6f.png)

## Describe Alternatives
For the annotation proposal: keep it as is

For the methods:
```kotlin
inline fun <K, V, reified T> DgsDataFetchingEnvironment.getDataLoader(): DataLoader<K, V> {
    return this.getDataLoader(T::class.java)
}
```

would work instead of 4 seperate functions, but this does not do any type checking, it merely avoids having to use `::class.java`:
![afbeelding](https://user-images.githubusercontent.com/30464310/152649411-54836e95-830c-4138-a8e3-770aacf15956.png)

## Contributing
Would be willing to contribute a PR, or two separate ones.

**Comment by srinivasankavitha** [comment]
@jord1e - What exactly is the issue? This should work fine in your case, no?
```
val dataloader1 = dfe.getDataLoader<String, Try<Person>>(PersonByIdDataLoader::class.java)
```

The above should anyway handle the case where the data loader is a mapped data loader. Why do you need the second one shown in your example, i.e.
```
val dataloader2 = dfe.getMappedDataLoader<String, Try<Person>>()
```

**Comment by jord1e** [comment]
> What exactly is the issue? This should work fine in your case, no?

Yes, it *works*, but it does not perform type-checking at *compile* time. Adding these utility methods gives library consumers to implement compile-time type-checking.

In the first example `PersonByIdDataLoader` is a `MappedBatchLoader<String, Try<Person>>`.
Because I am calling `getMappedBatchLoader<String, Person, PersonByIdDataLoader>()` in the example, the two first types are checked against the `PersonByIdDataLoader`, because the class returns `Try<Person>` we get a compilation error.

This is handy for when a return type or dataloader generic parameters changes types.

The four methods will be supplementary to the existing getDataLoader method.

**Comment by paulbakker** [comment]
This is merged in https://github.com/Netflix/dgs-framework/pull/1008 correct? Or is there more to it? 

Can this issue be closed?

**Comment by Emily** [comment]
Appears to have been solved by #904.

**Comment by jord1e** [comment]
It has indeed, I forgot to reply to Paul, thanks
